src/network/networkd-sysctl.c

```diff
@@ -534,6 +543,83 @@ int link_set_ipv6_mtu(Link *link, int log_level) {
         return 0;
 }
 
+static int ipv6_mtu_wait_synced_handler(sd_event_source *s, uint64_t usec, void *userdata);
+
+static int link_set_ipv6_mtu_async_impl(Link *link) {
+        uint32_t current_mtu;
+        int r;
+
+        assert(link);
+
+        /* When the link MTU is updated, it seems that the kernel IPv6 MTU of the interface is asynchronously
+         * reset to the link MTU. Hence, we need to check if it is already reset, and wait for a while if not. */
+
+        if (++link->ipv6_mtu_wait_trial_count >= 10) {
+                log_link_debug(link, "Timed out waiting for IPv6 MTU being synced to link MTU, proceeding anyway.");
+                r = link_set_ipv6_mtu(link, LOG_INFO);
+                if (r < 0)
+                        return r;
+
+                return 1; /* done */
+        }
+
+        /* Check if IPv6 MTU is synced. */
+        r = sysctl_read_ip_property_uint32(AF_INET6, link->ifname, "mtu", &current_mtu);
+        if (r < 0)
+                return log_link_warning_errno(link, r, "Failed to read IPv6 MTU: %m");
+
+        if (current_mtu == link->mtu) {
+                 /* Already synced. Update IPv6 MTU now. */
+                r = link_set_ipv6_mtu(link, LOG_INFO);
+                if (r < 0)
+                        return r;
+
+                return 1; /* done */
+        }
+
+        /* If not, set up a timer event source. */
+        r = event_reset_time_relative(
+                        link->manager->event, &link->ipv6_mtu_wait_synced_event_source,
+                        CLOCK_BOOTTIME, 100 * USEC_PER_MSEC, 0,
+                        ipv6_mtu_wait_synced_handler, link,
+                        /* priority = */ 0, "ipv6-mtu-wait-synced", /* force = */ true);
+        if (r < 0)
+                return log_link_warning_errno(link, r, "Failed to configure timer event source for waiting for IPv6 MTU being synced: %m");
```

Comment:
hmm, uh, a timer? doesn't the kernel send another netlink msg when it is done updating the mtu?

Reply:
It seems, at least judging from the log. Though, I've not checked the kernel code yet.